### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by jbabyhacker**

### CODE REVIEW

**Comment by jbabyhacker** [comment]
@nwotero please review

**Comment by nwotero** [comment]
Some of us have lives ya know :')

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+/// Recursive helper function to obtain the path to the crate root.
+fn walk_up(path: &path::Path) -> Option<path::PathBuf> {
+    let mut found_target = false;
+    let mut found_toml = false;
+
+    if path.is_dir() {
+        if let Ok(p) = fs::read_dir(&path) {
+            for q in p {
```
Comment (nwotero) [inline src/lib.rs]: One letter variable names are bad practice

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                if let Ok(r) = &q {
+                    if !found_target && r.file_name() == "target" {
+                        found_target = true;
+                    }
+
+                    if !found_toml && r.file_name() == "Cargo.toml" {
+                        found_toml = true;
+                    }
```
Comment (nwotero) [inline src/lib.rs]: If found_target and found_toml are both true, this loop should break

**Pull request merged**